### PR TITLE
ci: avoid to modify go.mod file

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

disable modules before to install `govendor`

## Why is it important?

`go get` modify the `go.mod` file so it breaks the `make check` target.

## Related issues

- Relates https://github.com/elastic/beats/issues/16840

### CODE REVIEW

**Comment by mikemadden42** [comment]
I wonder if we need `govendor` at all since we've moved to Go modules in `master` and `7.x`.

If we still need `govendor`, this fix looks correct though.

@kvch recommended just deleting the entire `go get` line.

File: .ci/scripts/install-go.sh [hunk 1]
```diff
@@ -16,4 +16,5 @@ chmod +x "${GVM_CMD}"
 gvm ${GO_VERSION}|cut -d ' ' -f 2|tr -d '\"' > ${PROPERTIES_FILE}
 
 eval $(gvm ${GO_VERSION})
-go get -u github.com/kardianos/govendor
+# see https://github.com/golang/go/issues/27643
+GO111MODULE=off go get -u github.com/kardianos/govendor
```
Comment (kuisathaverat) [inline .ci/scripts/install-go.sh]:
```suggestion
# GO111MODULE=off go get -u github.com/kardianos/govendor
```

**Comment by kuisathaverat** [comment]
there are no traces of govendor in logs, but there are some make target that were not executed

**Comment by urso** [comment]
we should not require govendor anymore. If it is still somewhere in the makefiles/scripts, then this might be an oversight. Go modules is a complete replacement.

File: .ci/scripts/install-tools.bat [hunk 2]
```diff
@@ -12,5 +12,5 @@ IF ERRORLEVEL 1 (
  curl -sL -o %WORKSPACE%\bin\gvm.exe https://github.com/andrewkroh/gvm/releases/download/v0.2.1/gvm-windows-amd64.exe
 )
 FOR /f "tokens=*" %%i IN ('"gvm.exe" use %GO_VERSION% --format=batch') DO %%i
-
+ste GO111MODULE=off
```
Comment (mikemadden42) [inline .ci/scripts/install-tools.bat]: Is this a typo?  Should `ste` be `set`?

Comment (kuisathaverat) [reply .ci/scripts/install-tools.bat]:
```suggestion
set GO111MODULE=off
```

File: .ci/scripts/install-go.sh [hunk 3]
```diff
@@ -16,4 +16,5 @@ chmod +x "${GVM_CMD}"
 gvm ${GO_VERSION}|cut -d ' ' -f 2|tr -d '\"' > ${PROPERTIES_FILE}
 
 eval $(gvm ${GO_VERSION})
-go get -u github.com/kardianos/govendor
+# see https://github.com/golang/go/issues/27643
+# GO111MODULE=off go get -u github.com/kardianos/govendor
```
Comment (kuisathaverat) [inline .ci/scripts/install-go.sh]:
```suggestion
```

File: .ci/scripts/install-tools.bat [hunk 4]
```diff
@@ -12,5 +12,5 @@ IF ERRORLEVEL 1 (
  curl -sL -o %WORKSPACE%\bin\gvm.exe https://github.com/andrewkroh/gvm/releases/download/v0.2.1/gvm-windows-amd64.exe
 )
 FOR /f "tokens=*" %%i IN ('"gvm.exe" use %GO_VERSION% --format=batch') DO %%i
-
+set GO111MODULE=off
```
Comment (kvch) [inline .ci/scripts/install-tools.bat]: FYI: There is no need to set `GO111MODULE` to `off`. Just change the next line to `go install -mod=vendor github.com/magefile/mage`.

File: .ci/scripts/install-tools.bat [hunk 5]
```diff
@@ -12,5 +12,5 @@ IF ERRORLEVEL 1 (
...
 go install github.com/elastic/beats/vendor/github.com/magefile/mage
```
Comment (kuisathaverat) [inline .ci/scripts/install-tools.bat]:
```suggestion
go install -mod=vendor github.com/magefile/mage
```

**Pull request merged**